src/modules/email/mentionEmail.js

```diff
@@ -2,68 +2,49 @@
 import jwt from 'jsonwebtoken';
 import * as pg from '../../lib/pg';
 import send from './sendEmail';
+import promisify from '../../lib/promisify';
 import handlebars from 'handlebars';
-import getMailObj from './buildMailObj';
 const MENTION_INTERVAL = 60 * 60 * 1000, MENTION_DELAY = 60 * 60 * 1000,
 	template = handlebars.compile(fs.readFileSync(__dirname + '/../../../templates/' +
-	config.app_id + '.digest.hbs', 'utf-8').toString()), log = new Logger(__filename),
-	connStr = config.connStr, conf = config.email, counter = new Counter();
+	config.app_id + '.mention.hbs', 'utf-8').toString()), log = new Logger(__filename, 'mention'),
+	connStr = config.connStr, conf = config.email;
 
 let lastEmailSent, end, i = 0;
 
 function initMailSending (userRel) {
+	new Promise(res => {
+		const user = userRel.puser;
+		log.info('Sending to : ', user);
+		if (!user.identities || !Array.isArray(user.identities)) {
+			log.info('No identities found for user: ', user);
+			return;
+		}
 
-	const user = userRel.currentUser;
-
-	if (!user.identities || !Array.isArray(user.identities)) {
-		log.info('No identities found for user: ', user);
-		return;
-	}
-
-	const rels = userRel.currentRels,
-		mailIds = user.identities.filter((el) => {
-			return /mailto:/.test(el);
-		});
-
-	mailIds.forEach((mailId) => {
-		counter.inc();
-		const emailAdd = mailId.slice(7);
+		const rels = userRel.prels,
+			mailIds = user.identities.filter((el) => {
+				return /mailto:/.test(el);
+			});
+			if(mailIds.length === 0) return;
+		const emailAdd = mailIds[0].slice(7);
+		const date = new Date().getDate(),
+			month = new Date().getMonth()+1,
+			year = new Date().getFullYear();
+ 			const formattedDate = date + '-' + month + '-' + year;
 		const emailHtml = template({
 				token: jwt.sign({ email: emailAdd }, conf.secret, { expiresIn: '5 days' }),
 				domain: config.server.protocol + '//' + config.server.host,
-				link : '&utm_content=' + encodeURIComponent(emailAdd) + '&utm_campaign=' + Date.now(),
+				link : '&utm_content=' + encodeURIComponent(user.id) + '&utm_campaign=' + encodeURIComponent(formattedDate),
 				rooms: rels,
-			}),
-			emailSub = `You h've been mentioned`;
-			// console.log("rels[0].threads: ", rels.length)
-		send(conf.from, emailAdd, emailSub, emailHtml, (e) => {
-			if (!e) {
-				log.info('Mention email successfully sent');
-				counter.dec();
-			} else {
-				counter.err(e);
-			}
-		});
-		counter.then((e) => {
-			if (e) {
-				log.error(e);
-				return;
-			}
-			pg.write(connStr, [ {
-				$: 'UPDATE jobs SET lastrun=&{end} WHERE id=&{jid}',
-				end,
-				jid: Constants.JOB_EMAIL_MENTION,
-			} ], (error) => {
-				lastEmailSent = end;
-				log.info('Mention email sent to ', i, ' users');
-				if (!error) log.info('successfully updated jobs for mention email');
 			});
+		const emailSub = `You have been mentioned`;
+		const sendMail = promisify(send.bind());
+		sendMail(conf.from, emailAdd, emailSub, emailHtml).then(() => {
```

Comment:
You can just return `sendMail(conf.from, emailAdd, emailSub, emailHtml)` without having to call `res` inside it's then callback. You can remove the promise wrapper entirely and don't even have to use an async function.